File: AbetOrder/AbetOrder/AbetOrder.Entity/DictionaryConfig.cs
﻿using OpenDataAccess.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AbetOrder.Entity
{
    [TableProperty(Name = "t_d_Dictionary", PrimaryKey = "Id", NoSelectNames = "IsDelete")]
    public class DictionaryConfig : EntityModel, IEntity
    {
        /// <summary> 
        /// 主键
        /// </summary> 
        public Guid Id { get; set; }
        /// <summary> 
        /// 名称
        /// </summary> 
        public string Name { get; set; }
        /// <summary> 
        /// 值
        /// </summary> 
        public string Value { get; set; }
        /// <summary> 
        /// 备注
        /// </summary> 
        public string Remark { get; set; }
        /// <summary> 
        /// 逻辑删除，1：删除,0:正常
        /// </summary> 
        public byte IsDelete { get; set; }
        /// <summary> 
        /// 创建时间
        /// </summary> 
        public DateTime CreateDate { get; set; }

        public long RowVersion { get; set; }
    }

    [TableProperty(Name = "v_Dictionary", PrimaryKey = "Id")]
    public class ViewDictionaryConfig : DictionaryConfig
    {
       
    }
}

File: testWeb618/configs/entities/DictionaryConfig.js
module.exports = {
    Name: "DictionaryConfig",
    PrimaryKey: "DictionaryConfigId",
    Properties: GetProperties()
  }
  
  function GetProperties() {
    return [
        GetProperty("DictionaryConfigId", "DictionaryConfigId"),
        GetProperty("Name", "键名"),
        GetProperty("Value", "值"),
        GetProperty("TypeName", "类型名"),
        GetProperty("Remark", "备注"),
        GetProperty("CreateDate", "创建时间")
    ]
  }
  
  function GetProperty(Name, Label) { return { Name, Label } }